graders/c/cgrader/cgrader.py

```diff
@@ -104,23 +120,16 @@ def run_command(
     ) -> str:
         if isinstance(command, str):
             command = shlex.split(command)
-        if sandboxed:
-            command = [
-                "su",
-                SB_USER,
-                "-s",
-                "/bin/bash",
-                "-c",
-                shlex.join(["PATH=" + self.path, *command]),
-            ]
-
         try:
+            if env is None:
+                env = {}
             proc = subprocess.Popen(
                 command,
-                env=env,
                 stdin=subprocess.PIPE,
                 stdout=subprocess.PIPE,
                 stderr=subprocess.STDOUT,
+                env={**os.environ, **({"PATH": self.path} if sandboxed else {}), **env},
```

Comment:
Need to test this further, but this should be a more modern and clean way to handle setting the PATH as well as running as the sandbox user.

Reply:
I'd be more comfortable if this change is done separately from the catch2 implementation, so it's properly tested in its own merits. For example, the new version does not run the command under bash, which could be an issue if the command includes bash-specific constructs.